drivers/ethernet/eth_xilinx_axienet.c

```diff
@@ -0,0 +1,654 @@
+		ret |= ETHERNET_HW_TX_CHKSUM_OFFLOAD;
+	}
+
+	return ret;
+}
+
+static const struct device *xilinx_axienet_get_phy(const struct device *dev)
+{
+	const struct xilinx_axienet_config *config = dev->config;
+
+	return config->phy;
+}
+
+static int xilinx_axienet_get_config(const struct device *dev, enum ethernet_config_type type,
+				     struct ethernet_config *config)
+{
+	const struct xilinx_axienet_config *dev_config = dev->config;
+	const struct xilinx_axienet_data *data = dev->data;
+	struct phy_link_state link_state;
+	int err;
+
+	switch (type) {
+	case ETHERNET_CONFIG_TYPE_AUTO_NEG:
+		/* enabled per default */
+		config->auto_negotiation = true;
+		return 0;
+	case ETHERNET_CONFIG_TYPE_DUPLEX:
+		err = phy_get_link_state(dev_config->phy, &link_state);
+		if (err != 0) {
+			LOG_ERR("Failed to get link state: %d", err);
+			return err;
+		}
+		config->full_duplex = link_state.is_up && (link_state.speed & LINK_FULL_10BASE_T ||
+							   link_state.speed & LINK_FULL_100BASE_T ||
+							   link_state.speed & LINK_FULL_1000BASE_T);
+		return 0;
+	case ETHERNET_CONFIG_TYPE_LINK:
+		err = phy_get_link_state(dev_config->phy, &link_state);
+		if (err != 0) {
+			LOG_ERR("Failed to get link state: %d", err);
+			return err;
+		}
+		if (!link_state.is_up) {
+			LOG_WRN("Ethernet is not up!");
+			return -EAGAIN;
+		}
+		if (link_state.speed & LINK_HALF_10BASE_T ||
+		    link_state.speed & LINK_FULL_10BASE_T) {
+			config->l.link_10bt = true;
+		}
+		if (link_state.speed & LINK_HALF_100BASE_T ||
+		    link_state.speed & LINK_FULL_100BASE_T) {
+			config->l.link_100bt = true;
+		}
+		if (link_state.speed & LINK_HALF_1000BASE_T ||
+		    link_state.speed & LINK_FULL_1000BASE_T) {
+			config->l.link_1000bt = true;
+		}
+		return 0;
+	case ETHERNET_CONFIG_TYPE_MAC_ADDRESS:
+		memcpy(config->mac_address.addr, data->mac_addr, sizeof(data->mac_addr));
+		return 0;
+	case ETHERNET_CONFIG_TYPE_PROMISC_MODE:
+		/* not supported yet */
+		config->promisc_mode = false;
+		return 0;
+	case ETHERNET_CONFIG_TYPE_RX_CHECKSUM_SUPPORT:
+		if (dev_config->have_rx_csum_offload) {
+			config->chksum_support =
+				ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP |
+				ETHERNET_CHECKSUM_SUPPORT_IPV6_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP;
+		} else {
+			config->chksum_support = ETHERNET_CHECKSUM_SUPPORT_NONE;
+		}
+		return 0;
+	case ETHERNET_CONFIG_TYPE_TX_CHECKSUM_SUPPORT:
+		if (dev_config->have_tx_csum_offload) {
+			config->chksum_support =
+				ETHERNET_CHECKSUM_SUPPORT_IPV4_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP |
+				ETHERNET_CHECKSUM_SUPPORT_IPV6_HEADER |
+				ETHERNET_CHECKSUM_SUPPORT_TCP | ETHERNET_CHECKSUM_SUPPORT_UDP;
+		} else {
+			config->chksum_support = ETHERNET_CHECKSUM_SUPPORT_NONE;
+		}
+		return 0;
+	default:
+		LOG_ERR("Unsupported configuration queried: %u", type);
+		return -EINVAL;
+	}
+}
```

Comment:
Please take a look into the ``ethernet_set_config()`` of subsys/net/l2/ethernet/ethernet_mgmt.c
There you find the supported defines for `get_config` when it is used via the iface. The others should be removed. If you want to get the speed you can get it via the phy

Reply:
I meant the ``ethernet_get_config()`` function.

I wanted to say, that some of the ``enum ethernet_config_type`` are in the overlaying api (``net_mgmt()``) for the iface are only implemented for get or set and not for both. For example the mac address is already in the iface and could be get from there, so you only should implement it in the set. 

also ``ETHERNET_CONFIG_TYPE_LINK`` and ``ETHERNET_CONFIG_TYPE_DUPLEX`` are mainly meant for the case where autonegation is not used. The correct way to get the current speed is via the phy. Thats also how it is done in the net iface shell. Frist get the phy dev via `net_eth_get_phy()` from the iface and the `phy_get_link_state()`.